pool.go

```diff
@@ -363,28 +377,45 @@ func (p *pool) doFetch(ctx context.Context, from string, c cid.Cid, attempt int)
 			fetchSpeedMetric.Observe(float64(received) / durationSecs)
 			fetchSizeMetric.Observe(float64(received))
 		}
+
+		if respHeader != nil {
+			saturnNodeId = respHeader.Get(saturnNodeIdKey)
+			saturnTransferId = respHeader.Get(saturnTransferIdKey)
+
+			cacheHit := respHeader.Get(saturnCacheHitKey)
+			if cacheHit == saturnCacheHit {
+				isCacheHit = true
```

Comment:
:thought_balloon:  is the idea here to use this and TTFBMS to identify fraud when L1 lied it did the work, but in fact it did no work and returned the thing it already had cached?

Reply:
@guanzo Can you take a look at this comment ? Why does the L1 need this field ?

Reply:
The idea here is to report if a request was a cache hit or not. This is useful for metrics and payment calculations. What lidel described is a more advanced fraud detection use case that isn't implemented currently

Reply:
Let's continue in https://filecoinproject.slack.com/archives/C031K3BG527/p1677088893305269 ;-)